@MyDaq/MyDaq.m

```diff
@@ -488,40 +421,60 @@ function cursorButtonCallback(this, hObject, ~)
         %Callback for the instrument menu
         function instrMenuCallback(this,hObject,~)
             val=hObject.Value;
-            names=hObject.String;
-            %Finds the correct instrument tag as long as an instrument is
-            %selected
-            if val~=1
-                try 
-                    tag=getTag(this,names(val));
-                catch 
-                    error('Invalid instrument selected: %s',names{val});
-                end
+            if val==1
+                return
+            else
+                tag = hObject.ItemsData{val};
             end
             
-            %If instrument is valid and not open, opens it. If it is valid
-            %and open it changes focus to the instrument control window.
-            if (ismember(tag,this.instr_tags) && ...
-                    ~ismember(tag,this.open_instrs))
-                openInstrument(this,tag);
-            elseif ismember(tag,this.open_instrs)
-                if isfield(this.InstrApps,tag)
-                    prop_names=properties(this.InstrApps.(tag));
-                    figure_ind=cellfun(@(x) isa(this.InstrApps.(tag).(x),...
-                        'matlab.ui.Figure'),prop_names);
-                    fig_handle=this.InstrApps.(tag).(prop_names{figure_ind});
+            % Run-files themselves are supposed to prevent duplicated
+            % instances, but let DAQ handle it as well for safety
+            if ismember(tag, this.running_progs)
+                % Change focus to the instrument control window
+                fig_handle = findfigure(this.RunningPrograms.(tag));
+                if isempty(fig_handle)
+                    % If unable, try the same for .Gui object inside
+                    try
+                        fig_handle =...
+                            findfigure(this.RunningPrograms.(tag).Gui);
+                    catch

```

Comment:
Maybe we can put a warning here, e.g. `Program %s shows as open, but its GUI could not be found`.